pkg/pillar/cmd/volumemgr/updatestatusold.go

```diff
@@ -0,0 +1,371 @@
+				status.VolumeID, status.BlobSha256)
+		} else {
+			log.Infof("lookForVerifiedOld: Found PersistImageStatus: %s based on ImageSha256 %s VolumeID %s",
+				status.DisplayName, status.BlobSha256, status.VolumeID)
+			if !status.DownloadOrigin.HasPersistRef {
+				log.Infof("lookForVerifiedOld: Adding PersistImageStatus reference for VolumeStatus: %s", status.BlobSha256)
+				AddOrRefCountPersistImageStatus(ctx, ps.Name, ps.ObjType, ps.FileLocation, ps.ImageSha256, ps.Size)
+				status.DownloadOrigin.HasPersistRef = true
+				changed = true
+			}
+			//Marking the VolumeStatus state as VERIFIED as we already have a PersistImageStatus for the volume
+			if status.State != types.VERIFIED {
+				status.State = types.VERIFIED
+				status.Progress = 100
+				changed = true
+			}
+			if status.FileLocation != ps.FileLocation {
+				status.FileLocation = ps.FileLocation
+				log.Infof("lookForVerifiedOld: Update FileLocation for %s: %s",
+					status.Key(), status.FileLocation)
+				changed = true
+			}
+			// If we don't already have a RefCount add one
+			if !status.DownloadOrigin.HasVerifierRef {
+				log.Infof("!HasVerifierRef")
+				// We don't need certs since Status already exists
+				MaybeAddVerifyImageConfigOld(ctx, *status, false)
+				status.DownloadOrigin.HasVerifierRef = true
+				changed = true
+			}
+			//Wait for VerifyImageStatus to appear
+			return nil, changed
+		}
+	} else {
+		log.Infof("Found %s based on VolumeID %s sha %s",
+			status.DisplayName, status.VolumeID, status.BlobSha256)
+		// If we don't already have a RefCount add one
+		// No need to checkCerts since we have a VerifyImageStatus
+		c := kickVerifierOld(ctx, status, false)
+		if c {
+			changed = true
+		}
+		return vs, changed
+	}
+	return vs, changed
+}
+
+// doDelete returns changed boolean
+// XXX need return value?
+func doDelete(ctx *volumemgrContext, status *types.OldVolumeStatus) bool {
+	changed := false
+
+	// XXX support other types
+	if status.Origin == types.OriginTypeDownload {
+		if status.DownloadOrigin.HasDownloaderRef {
+			MaybeRemoveDownloaderConfig(ctx, status.ObjType,
+				status.BlobSha256)
+			status.DownloadOrigin.HasDownloaderRef = false
+			changed = true
+		}
+		if status.DownloadOrigin.HasVerifierRef {
+			MaybeRemoveVerifyImageConfig(ctx, status.ObjType,
+				status.BlobSha256)
+			status.DownloadOrigin.HasVerifierRef = false
+			changed = true
+		}
+		if status.DownloadOrigin.HasPersistRef {
+			ReduceRefCountPersistImageStatus(ctx, status.ObjType, status.BlobSha256)
+			status.DownloadOrigin.HasPersistRef = false
+			changed = true
+		}
+	}
+
+	if status.VolumeCreated {
+		// Asynch destruction; make sure we have a request for the work
+		MaybeAddWorkDestroyOld(ctx, status)
+		vr := lookupVolumeWorkResult(ctx, status.Key())
+		if vr != nil {
+			log.Infof("VolumeWorkResult(%s) location %s, created %t",
+				status.Key(), vr.FileLocation, vr.VolumeCreated)
+			deleteVolumeWorkResult(ctx, status.Key())
```

Comment:
Shouldn't this be calling the Old function?

Reply:
Sorry for my confusion

Reply:
No issues.